stratify/__init__.py

```diff
@@ -1,11 +1,3 @@
-from __future__ import absolute_import
-
-from ._vinterp import (interpolate, interp_schemes,  # noqa: F401
-                       extrap_schemes, INTERPOLATE_LINEAR, INTERPOLATE_NEAREST,
-                       EXTRAPOLATE_NAN, EXTRAPOLATE_NEAREST,
-                       EXTRAPOLATE_LINEAR, PyFuncExtrapolator,
-                       PyFuncInterpolator)
 from ._bounded_vinterp import interpolate_conservative  # noqa: F401
-
-
-__version__ = '0.2.dev0'
+from ._version import version as __version__  # noqa: F401
```

Comment:
ok, so `setuptools_scm` is taking the version number from the latest tag and then applying logic to calculate a new one?  Sorry, this is all new to me, I'm sure it makes sense once you've done it a couple of times but seems like magic! If I was to download the package without the `.git` repo, would it still be able to build/install from source?

Reply:
Yup, you've essentially got it. This following section in the `pyproject.toml` file is like a nominated scratch area for `setuptools-scm`:
```
[tool.setuptools_scm]
write_to = "stratify/_version.py"
```
It can be anything that you want. Also, you can control how `scm` generates the offset from the latest tag, but what you get out the box is good enough for me.

Note that, there is also this plugin [setuptools-scm-git-archive](https://pypi.org/project/setuptools-scm-git-archive/) which might address your question, but I think the use case to using the source code outside `git` is a bit rare - particularly when it comes to packaging IMHO.

Reply:
@jamesp I actually stumbled across the actual answer to your question regarding the necessity to have access to the `.git` repo... as I had to jump through that hoop when attempting to build the `conda-forge` package for the `0.2.post0` release.

You do need access to a `.git` repo and the associated artifacts for `setuptools-scm` to do its job of fathoming out what the version of the package. However, if that's not possible, for example this is exactly the case when attempting to `conda-build` your package from a `.tar.gz` archive from GitHub, then you need to instead hit the associated PyPI URL for your package as this contain the necessary metadata for `setuptools-scm` to do it's job i.e., see https://github.com/conda-forge/python-stratify-feedstock/blob/master/recipe/meta.yaml#L9.

Bingo. Just works :+1:

Reply:
Yep, this is the use-case I was thinking about.  Good to see there is a reasonable workaround, although having conda-forge hanging off of pypi does seem a little Rube Goldberg!